streams/src/main/java/org/apache/kafka/streams/processor/InferringConsumerRecordTimestampExtractor.java

```diff
@@ -38,37 +38,33 @@
  * using this extractor effectively provides <i>ingestion-time</i> semantics.
  * <p>
  * If you need <i>processing-time</i> semantics, use {@link WallclockTimestampExtractor}.
- * <p>
- * If a record has a negative (invalid) timestamp value a new timestamp will be inferred as the current
- * Streams time.
  *
- * @see ConsumerRecordTimestampExtractor
- * @see RobustConsumerRecordTimestampExtractor
+ * @see FailingConsumerRecordTimestampExtractor
+ * @see SkipInvalidConsumerRecordTimestampExtractor
  * @see WallclockTimestampExtractor
  */
-public class InferringConsumerRecordTimestampExtractor implements TimestampExtractor {
+public class InferringConsumerRecordTimestampExtractor extends AbstractConsumerRecordTimestampExtractor {
```

Comment:
What about a name like "ReplaceInvalidTimestampWithCurrentStreamTime"?  This suggestion is still a bit too long IMHO, but it is much more descriptive (I think) than "InferringConsumerRecordTimestampExtractor".